### PR TITLE
Added Templates Subscription

### PR DESCRIPTION
**Pull request by cheina97**
Components modified

- TemplateTableLogic

This PR add subscription to the query used to get templates in a workspace.

### CODE REVIEW

File: frontend/src/components/workspaces/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -30,7 +29,6 @@ const Dashboard: FC<IDashboardProps> = ({ ...props }) => {
             }))}
             onClick={(id: number) => {
```
Comment (QcFe) [inline frontend/src/components/workspaces/Dashboard/Dashboard.tsx:30]:
```suggestion
            onClick={setSelectedWs}
```
It should be possible to do something alike

File: frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx [hunk 2]
```diff
@@ -113,25 +110,80 @@ const TemplatesTableLogic: FC<ITemplateTableLogicProps> = ({ ...props }) => {
...
+              t =>
+                t?.metadata?.id ===
+                data?.updatedTemplate?.template?.metadata?.id
+            );
+            if (oldItem) {
+              if (JSON.stringify(oldItem) === JSON.stringify(newItem)) {
+                //template have been deleted
+                prev.templateList.templates = prev.templateList?.templates!.filter(
+                  t =>
+                    data?.updatedTemplate?.template?.metadata?.id !==
```
Comment (QcFe) [inline frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx:151]:
```suggestion
                    newItem.metadata?.id !==
```
Also below in another two or three occurrences

File: frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx [hunk 3]
```diff
@@ -113,25 +110,80 @@ const TemplatesTableLogic: FC<ITemplateTableLogicProps> = ({ ...props }) => {
...
+              subscriptionResult = 'added';
+            }
+          }
+
+          notification['info']({
+            message: newItem.spec?.name,
+            description: `This Template have been ${subscriptionResult}`,
+          });
+
+          const newData = { ...prev };
```
Comment (QcFe) [inline frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx:180]: Is it necessary to make a new object? In that case it couldn't it be more meaninful to create it earlier and work on it from the beginning?

File: frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx [hunk 4]
```diff
@@ -113,25 +110,80 @@ const TemplatesTableLogic: FC<ITemplateTableLogicProps> = ({ ...props }) => {
...
+                ...prev.templateList.templates,
+                data?.updatedTemplate?.template!,
+              ];
+              subscriptionResult = 'added';
+            }
+          }
+
+          notification['info']({
+            message: newItem.spec?.name,
+            description: `This Template have been ${subscriptionResult}`,
```
Comment (QcFe) [inline frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx:177]:
```suggestion
            message: 'Workspace update!',
            description: `Template <i>${newItem.spec?.name}</i> has been ${subscriptionResult}`,
```
I'd suggest something like this. Please check whether html tags are supported in the notification, if not just remove them.

**Review by QcFe — APPROVED** [review]
Looks good. I leave my usual style bits (if they make sense 🥴)

File: frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx (hunk 4 above)
Comment (QcFe) [reply frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx:177]: Don't know if @giorio94 has any better suggestion

Comment (giorio94) [reply frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx:177]: Not really. Could just specify the workspace in the message, if that makes sense.

File: frontend/src/components/workspaces/Dashboard/Dashboard.tsx (hunk 1 above)
Comment (cheina97) [reply frontend/src/components/workspaces/Dashboard/Dashboard.tsx:30]: Did it

File: frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx (hunk 2 above)
Comment (cheina97) [reply frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx:151]: Done

File: frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx (hunk 3 above)
Comment (cheina97) [reply frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx:180]: Done

**Comment by QcFe** [comment]
@cheina97 I just realized I wrote `Workspace update!` instead of `Workspace updated!` in the message, could you please make this small change since the merge action seems to have broken? 🥴

**Comment by giorio94** [comment]
> @cheina97 I just realized I wrote `Workspace update!` instead of `Workspace updated!` in the notification message, could you please make this last small change since the merge action seems to have broken? woozy_face

GitHub actions do not seem to be in good shape this afternoon

**Comment by cheina97** [comment]
No problem, i've noticed also some bugs when route changes. I'm going to include a patch in this pr for this and the other subscriptions. I'll wait to squash the commit with those fixes.

File: frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx [hunk 5]
```diff
@@ -119,13 +114,14 @@ const TemplatesTableLogic: FC<ITemplateTableLogicProps> = ({ ...props }) => {
...
     onCompleted: setDataTemplate,
+    fetchPolicy: 'network-only',
   });
 
   const { instances } = dataInstances?.instanceList ?? {};
 
   useEffect(() => {
     if (!loadingTemplate) {
-      subscribeToMoreTemplates({
+      const unsubscribeToMoreTemplates = subscribeToMoreTemplates({
```
Comment (giorio94) [inline frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx:124]: Is *unsubscribe* correct?

File: frontend/src/components/workspaces/DashboardLogic/DashboardLogic.tsx [hunk 6]
```diff
@@ -12,8 +12,8 @@ const DashboardLogic: FC<{}> = () => {
 
   const { loading, error, subscribeToMore } = useTenantQuery({
     variables: { tenantId: userId ?? '' },
-    notifyOnNetworkStatusChange: true,
     onCompleted: setData,
+    fetchPolicy: 'network-only',
```
Comment (giorio94) [inline frontend/src/components/workspaces/DashboardLogic/DashboardLogic.tsx:16]: Nit. Could be probably defined as constant, if that makes sense to you.

**Review by giorio94 — APPROVED** [review]
From my side you can go on and squash the commits...

File: frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx (hunk 5 above)
Comment (cheina97) [reply frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx:124]: Yes, but then i didn't use it. I'm going to remove it

File: frontend/src/components/workspaces/DashboardLogic/DashboardLogic.tsx (hunk 6 above)
Comment (cheina97) [reply frontend/src/components/workspaces/DashboardLogic/DashboardLogic.tsx:16]: For typescript "network-only" is not a normal string but a special type that can have only a list of value, there is an implicit cast. Anyway is not a problem to declare it as a constant. i'm going to do it

**Pull request merged**